src/main/kotlin/com/atlassian/performance/tools/dockerinfrastructure/jira/SetUpFromScratchAction.kt

```diff
@@ -128,19 +130,40 @@ internal class SetUpFromScratchAction(
         for (i in 1..10) {
             val closeButtons = driver
                 .findElements(By.className("icon-close"))
-                .filter { it.isDisplayed }
+                .filter { it.isDisplayed && it.isEnabled }
             if (closeButtons.isEmpty()) {
                 break
             } else {
-                closeButtons.first().click()
+                try {
+                    closeButtons.first().click()
+                } catch (e: ElementNotInteractableException) {
+                }
                 Thread.sleep(slideOutTime.toMillis() * 2)
             }
         }
     }
 
     private fun waitAndClick(by: By, timeout: Duration = Duration.ofMinutes(5)) {
-        driver.wait(timeout, ExpectedConditions.visibilityOfElementLocated(by))
-        driver.findElement(by).click()
+        driver.wait(timeout, ExpectedConditions.elementToBeClickable(by))
+            .click()
+    }
+    
+    private fun clickAndAwaitTransition(by: By, timeout: Duration = Duration.ofMinutes(5)) {
+        val clickable = driver.wait(timeout, ExpectedConditions.elementToBeClickable(by))
+        clickable.click()
+        val initialTimeout = Duration.ofSeconds(10)
+        try {
+            driver.wait(initialTimeout, ExpectedConditions.stalenessOf(clickable))
+        } catch (e: Exception) {
+            try {
+                //this both resends the click and executes a staleness check better than stalenessOf
+                clickable.click()
+            } catch (e: StaleElementReferenceException) {
```

Comment:
We usually stay away from exception-class-driven flow control. But I see you've been fighting the flakiness fiercely, so let's ship this fix!